### PR TITLE
asn1/ans_moid.c: overhaul do_create.

### PR DESCRIPTION
**Pull request by dot-asm**
Original could allocate nid and then bail out on malloc failure. Instead
allocate first *then* attempt to create object.

Triggered by #6982.

### CODE REVIEW

File: crypto/asn1/asn_moid.c [hunk 1]
```diff
@@ -60,29 +60,20 @@ void ASN1_add_oid_module(void)
...
 {
     int nid;
-    ASN1_OBJECT *oid;
     const char *ln, *ostr, *p;
-    char *lntmp;
+    char *lntmp = NULL;
+
     p = strrchr(value, ',');
-    if (!p) {
+    if (p != NULL) {
```
Comment (paulidale) [inline crypto/asn1/asn_moid.c]: Why has the check been reversed here?

Comment (dot-asm) [reply crypto/asn1/asn_moid.c]: Right! Typo! Thanks!

**Comment by dot-asm** [comment]
Oh! I accidentally committed unrelated change to bn/bn_x931.c. Removed...

**Comment by kroeckx** [comment]
Could you fix OBJ_create()'s manpage to say that it also return an NId or NID_undef on error?

**Pull request closed without merging**